pkg/agent/openflow/network_policy.go

```diff
@@ -2220,3 +2228,69 @@ func (f *featureNetworkPolicy) l7NPTrafficControlFlows() []binding.Flow {
 			Done(),
 	}
 }
+
+func (f *featureNetworkPolicy) loggingNPPacketFlowWithOperations(cookieID uint64, operations uint8) binding.Flow {
+	loggingOperations := binding.NewRegMark(PacketInOperationField, uint32(operations))
+	fb := L2ForwardingOutTable.ofTable.BuildFlow(priorityNormal).
+		MatchRegMark(SendToControllerRegMark, loggingOperations)
+	if f.ovsMetersAreSupported {
+		fb = fb.Action().Meter(PacketInMeterIDNP)
+	}
+	return fb.Action().SendToController([]byte{uint8(PacketInCategoryNP), operations}, false).
+		Cookie(cookieID).
+		Done()
+}
+
+func (f *featureNetworkPolicy) initLoggingFlows() []binding.Flow {
+	maxOperationValue := PacketInNPLoggingOperation + PacketInNPStoreDenyOperation + PacketInNPRejectOperation
+	flows := make([]binding.Flow, 0, maxOperationValue)
+	if !f.enableAntreaPolicy {
+		return flows
+	}
+	cookieID := f.cookieAllocator.Request(f.category).Raw()
+	for operation := 1; operation <= maxOperationValue; operation++ {
```

Comment:
why it needs a group per operation combination?

Reply:
The operation is used in OF action "controller(userdata)", which is consumed by the packetIn handler to identify packetIN type. It is a static configuration for OF mesage, so we can not use a single flow to expresss different operations (userdata values).

Reply:
Thanks for explanation.